File: 0704-binary-search/0704-binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(numsx, targetx)
    @nums=numsx
    def searchrec nums, target
        return -1 if nums.empty? || nums.length<2&&nums[0]!=target
        index=nums.length/2
        if nums[index]==target
            return @nums.index(target)
        elsif nums[index]<target
            p nums[index]
            return searchrec(nums[index+1..nums.length-1], target)
        else
            p nums[index]
            return searchrec(nums[0..index-1], target)
        end
    end
    return searchrec numsx, targetx
end

File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    /*
    left=0
    right=1
     */
    let arr=[...nums]
    let left=0
    let right=arr.length
    let mid=Math.floor((left+right)/2)//1
    while(left<=right){
        if(target==arr[mid]){ //arr[4]==-1
            return mid
        }else if(target>arr[mid]){//-1>arr[1] | 
            left=mid+1
            // mid=Math.ceil((left+right)/2)//1
        }else{
            right=mid-1 // // //
        }
            mid=Math.floor((left+right)/2)//1
    }
    return -1
};